### DISCUSSION TITLE
Exemple d'utilisation du web scrapper

### DISCUSSION THREAD

**Discussion by vonv** [FreshRSS/FreshRSS#4647] [Q&A] [answered]
Alors d'abord merci mais un énorme merci pour FreshRSS: le RSS ne mourra jamais ! :)

Et alors - bim - nouvelle fonctionnalité dont je rêve depuis des années: le web scrapping.

Par contre, si on pouvait avoir un exemple dans la doc de son utilisation, par exemple sur une page comme: https://github.com/FreshRSS/FreshRSS/releases

Ce serait le top ! Car là, je galère.

D'avance merci.

**Comment by Alkarex** [comment]
Si vous me donnez une page Web, je vous fais un exemple.

Exemple basique : https://github.com/FreshRSS/FreshRSS/pull/4220#issuecomment-1038450459

Autre exemple : https://github.com/FreshRSS/FreshRSS/pull/4220#issuecomment-1053472107

Exemple un peu plus compliqué https://github.com/FreshRSS/FreshRSS/pull/4220#issuecomment-1038399003

**Comment by vonv** [comment]
Merci beaucoup ! Super, les exemples.
Ce doit être mon flux qui a un souci. Je vais investiguer.

Merci.

**Comment by marienfressinaud** [comment, marked as answer]
Pour info, quelqu'un a également publié un article complet sur le sujet : https://danq.me/2022/09/27/freshrss-xpath/

**Reply by Frenzie** [reply]
Nice article. h-feed?

**Reply by aledeg** [reply]
My reaction exactly, then I followed the link in the article.

http://microformats.org/wiki/h-feed

**Reply by Alkarex** [reply]
Very nice article indeed :-)
Maybe someone could reach out to danq to get an idea of how frequent `h-feed` is on Web sites not already proposing an RSS/ATOM feed.

**Reply by Alkarex** [reply]
P.S. I have left a comment on https://danq.me/2022/09/27/freshrss-xpath/#comment-221140

**Reply by math-GH** [reply]
There is a Github repo for h-feed: https://github.com/microformats/h-feed
It looks like that the standard is still in progress/discussion

**Reply by math-GH** [reply]
Here are some example websites: https://github.com/microformats/h-feed/issues/9

**Reply by marienfressinaud** [reply]
I implemented micro-formats and h-feed on my own [blog](https://marienfressinaud.fr/blog.html) a few months ago. I also started a parser for [flusio](https://github.com/flusio/flusio) at this time, but I had some difficulties. I can't remember why I didn't finish though :/ I'll take a look this week and will share my thoughts.

**Reply by marienfressinaud** [reply]
It looks like SimplePie [already supports h-feed](https://github.com/FreshRSS/FreshRSS/blob/edge/lib/SimplePie/SimplePie/Parser.php#L476). I think it only needs the [php-mf2 parser](https://github.com/microformats/php-mf2) to be enabled (see https://github.com/FreshRSS/FreshRSS/blob/edge/lib/SimplePie/SimplePie.php#L1783).

I did a naive parser this morning (see https://github.com/flusio/flusio/pull/568/files), but I think I'll need the php-mf2 parser because it doesn't work very well.

**Reply by Frenzie** [reply]
> It looks like SimplePie [already supports h-feed](https://github.com/FreshRSS/FreshRSS/blob/edge/lib/SimplePie/SimplePie/Parser.php#L476).

Quickest fix ever. ;-D

**Reply by math-GH** [reply]
The commits into our SimplePie copy are already 5 years old :D

**Reply by Alkarex** [reply]
> The commits into our SimplePie copy are already 5 years old :D

The code that counts is https://github.com/FreshRSS/FreshRSS/tree/edge/lib/SimplePie and our repo https://github.com/FreshRSS/simplepie is mostly for sending PR upstream

**Reply by Alkarex** [reply]
Another nice article by @Dan-Q https://danq.me/2022/11/23/far-side-freshrss-xpath/

**Reply by Alkarex** [reply]
https://danq.me/2023/03/28/forward-comic-freshrss-xpath/

**Reply by thiswillbeyourgithub** [reply]
Let me add thta you can use the "inspect" feature of firefox, then right click on an element and "copy XPath"
